src/query/server/server.go

```diff
@@ -205,6 +210,11 @@ func Run(runOpts RunOptions) {
 
 	engine := executor.NewEngine(backendStorage, scope.SubScope("engine"))
 
+	// TODO: centralize application of defaults?
+	if cfg.Limits == nil {
```

Comment:
General pattern seems to be using a generator for the configs which is passed a cfg.Limits object which it then can decode with defaults as appropriate

Reply:
Also, definitely not going to add it here, but any thoughts on https://github.com/imdario/mergo? Seems like a natural fit for this kind of thing (and avoids all of the `if field == empty` boilerplate).

Reply:
Ah cool thanks, will go with that pattern.

Reply:
Doh nvm; misread you.

Reply:
Slightly broader discussion: any thoughts on introducing `base.yml` (or `defaults.yml`) files for services, and using `xconfig.LoadFiles` to do the merging? Things I like about that approach:

- Defaults are very explicitly defined, and in a central location (and thus easy to find).
- Avoids unnecessary boiler plate from getter methods or `if cfg == nil` checks throughout the code.

cc. @arnikola

Reply:
FYI: ended up making limits off by default (so as to not disrupt anybody's existing expensive queries), which had the side effect of simplifying the defaults. `<= 0` now implies no limit, which is hopefully intuitive.